File: src/main/java/com/example/FleetSystem/model/Invoice.java
package com.example.FleetSystem.model;

import lombok.*;
import javax.persistence.*;
import java.sql.Date;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.YearMonth;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Builder
@Entity
public class Invoice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String businessUnit;
    private String invoiceNumber;
    private Date invoiceDate;
    private YearMonth invoiceMonth;
    private String invoiceType;
    private String invoiceCategory;
    private String invoiceFrom;
    private String invoiceTo;
    private Float amountBeforeTax;
    private Float taxableAmount;
    private Float taxPercent;
    private Float VATAmount;
    private Float amountAfterVAT;
    private Long lineNumber;
    private String plateNumber;
    private String agreementNumber;
    private Integer monthlyRate;
    private Long supplierControlNumber;
    private Integer vendorAmount;
    private Date dateFrom;
    private Date dateTo;
    private Long vendorVehicleRefNumber;
    private Float lineAmountWithoutTax;
    private Float lineTaxRate;
    private Float lineTaxAmount;
    private Float lineAmountWithTax;
    private String uuid;
    private LocalDate createdAt;
    private LocalDate updatedAt;

    @ManyToOne
    private InvoiceFile invoiceFile;

    @ManyToOne
    private Vendor supplier;

    @ManyToOne
    @JoinColumn(name = "created_by")
    private User createdBy;

    @ManyToOne
    @JoinColumn(name = "updated_by")
    private User updatedBy;
}

File: finance-ui/fms-finance/src/app/modal/invoice.ts

export interface Invoice {
    id: number | null | undefined; 
    businessUnit: string | null | undefined;
    invoiceNumber: string | null | undefined;
    invoiceDate: Date | null | undefined; 
    invoiceMonth: string | null | undefined; 
    invoiceType: string | null | undefined;
    invoiceCategory: string | null | undefined;
    invoiceFrom: string | null | undefined;
    invoiceTo: string | null | undefined;
    amountBeforeTax: number | null | undefined; 
    taxableAmount: number | null | undefined;
    taxPercent: number | null | undefined;
    VATAmount: number | null | undefined;
    amountAfterVAT: number | null | undefined;
    lineNumber: number | null | undefined; 
    plateNumber: string | null | undefined;
    agreementNumber: string | null | undefined;
    monthlyRate: number | null | undefined; 
    supplierControlNumber: number | null | undefined; 
    vendorAmount: number | null | undefined; 
    dateFrom: Date | null | undefined; 
    dateTo: Date | null | undefined; 
    vendorVehicleRefNumber: number | null | undefined; 
    lineAmountWithoutTax: number | null | undefined;
    lineTaxRate: number | null | undefined;
    lineTaxAmount: number | null | undefined;
    lineAmountWithTax: number | null | undefined;
    uuid: string | null | undefined;
    fileName: string | null | undefined;
    createdAt: Date | null | undefined; 
    updatedAt: Date | null | undefined; 
    supplier: {
        id: Number| null | undefined
        vendorName: String| null | undefined
        officeLocation: String | null | undefined
        attachments: String | null | undefined
    }
  }
  